src/main/OP.ts

```diff
@@ -149,11 +149,46 @@ function createResponseOptsFromBuilderOrExistingOpts(opts: { builder?: OPBuilder
     ? {
         registration: {
           issuer: opts.builder.issuer,
+          authorizationEndpoint: opts.builder.responseRegistration.authorizationEndpoint,
+          tokenEndpoint: opts.builder.responseRegistration.tokenEndpoint,
+          userinfoEndpoint: opts.builder.responseRegistration.userinfoEndpoint,
+          jwksUri: opts.builder.responseRegistration.jwksUri,
+          registrationEndpoint: opts.builder.responseRegistration.registrationEndpoint,
+          scopesSupported: opts.builder.responseRegistration.scopesSupported,
+          responseTypesSupported: opts.builder.responseRegistration.responseTypesSupported,
+          responseModesSupported: opts.builder.responseRegistration.responseModesSupported,
+          grantTypesSupported: opts.builder.responseRegistration.grantTypesSupported,
+          acrValuesSupported: opts.builder.responseRegistration.acrValuesSupported,
+          subjectTypesSupported: opts.builder.responseRegistration.subjectTypesSupported,
+          idTokenSigningAlgValuesSupported: opts.builder.responseRegistration.idTokenSigningAlgValuesSupported,
+          idTokenEncryptionAlgValuesSupported: opts.builder.responseRegistration.idTokenEncryptionAlgValuesSupported,
```

Comment:
Hmzz, we probably want to do some refactoring and destructuring in here.

Reply:
yeah right now it's quite messy